spec.bs

```diff
@@ -232,7 +232,161 @@ to conduct an auction to select an advertisement to display to the user, they ca
 {{Window/navigator}}.{{Navigator/runAdAuction()}} function, providing an auction configuration that
 tells the browser how to conduct the auction and which on-device recorded interests are allowed to
 bid in the auction for the chance to display their advertisement.
-  
+
+<h3 id="script-runners">Script Runners</h3>
+
+<pre class="idl">
+[Exposed=InterestGroupBiddingScriptRunner]
+interface InterestGroupBiddingScriptRunner {
+};
+
+[Exposed=InterestGroupScoringScriptRunner]
+interface InterestGroupScoringScriptRunner {
+};
+
+[Exposed=InterestGroupReportingScriptRunner]
+interface InterestGroupReportingScriptRunner {
+};
+
+[Exposed=InterestGroupScriptRunner]
+interface InterestGroupScriptRunner {
+};
+</pre>
+
+An additional requirement to the interest group script runner globals defined in this specification
+is that they must not expose any interfaces from other specifications whose own exposure set is the
+special value "*". The only interfaces that can be exposed inside of the globals defined in this
+specification are those that explicitly list the global names provided here.
+
+<pre class="idl">
+[Exposed=InterestGroupBiddingScriptRunner,
+Global=(InterestGroupScriptRunner, InterestGroupBiddingScriptRunner)]
+interface InterestGroupBiddingScriptRunnerGlobalScope{
+  boolean setBid();
+  boolean setBid(GenerateBidOutput generateBidOutput);
+  undefined setPriority(double priority);
+  undefined setPrioritySignalsOverride(DOMString key, double priority);
+};
+
+[Exposed=InterestGroupScoringScriptRunner,
+Global=(InterestGroupScriptRunner, InterestGroupScoringScriptRunner)]
+interface InterestGroupScoringScriptRunnerGlobalScope{
+};
+
+[Exposed=InterestGroupReportingScriptRunner,
+Global=(InterestGroupScriptRunner, InterestGroupReportingScriptRunner)]
+interface InterestGroupReportingScriptRunnerGlobalScope{
+  undefined sendReportTo(DOMString url);
+};
+
+dictionary AdSize {
+  required double width;
+  required DOMString widthUnits;
+  required double height;
+  required DOMString heightUnits;
+};
+
+dictionary AdDescriptor {
+  required DOMString url;
+  AdSize adSize;
+};
+
+dictionary GenerateBidOutput {
+  required double bid;
+  required (DOMString or AdDescriptor) adDescriptor;
+  DOMString ad;
+  sequence<(DOMString or AdDescriptor)> adComponentDescriptors;
+  double adCost;
+  double modelingSignals;
+  boolean allowComponentAuction = false;
+};
+</pre>
+
+Issue: TODO: Turn the follwing methods into spec language.
+
+The setBid() method steps are:
+1. Clear the bid.
+1. Return true.
+
+The setBid(|generateBidOutput|) method steps are (TODO: Finish this. Some steps are not in spec

```

Comment:
I understand that these are not quite spec language yet, per the above "Issue", but maybe we can put these in `<div algorithm>`s at least.